tests/integration/conftest.py

```diff
@@ -162,16 +167,9 @@ async def deploy_model(
         )
     with ops_test.model_context(model_name) as the_model:
         async with ops_test.fast_forward():
-            await asyncio.gather(
-                *(
-                    the_model.deploy(**asdict(charm))
-                    for charm in deploy_args
-                    if charm.application_name not in the_model.applications
-                )
-            )
-            await the_model.integrate("k8s", "k8s-worker:cluster")
+            await the_model.deploy(bundle.render)
```

Comment:
pylibjuju knows how to deploy a bundle over an existing model and "do the right thing"

it will even upgrade any apps in the model with the fresh charms!! WOOT!